src/util/debug.c

```diff
@@ -350,7 +352,16 @@ void ldb_debug_messages(void *context, enum ldb_debug_level level,
     }
 
     if (DEBUG_IS_SET(loglevel)) {
-        sss_vdebug_fn(__FILE__, __LINE__, "ldb", loglevel, APPEND_LINE_FEED,
+        if (level == LDB_DEBUG_TRACE) {
+            /* This is most probably `tevent_debug()` which has own '\n'.
+               As this log level should be virtually never enabled
+               performance considerations (strlen) are not that important.
+             */
+            if (fmt[strlen(fmt)-1] == '\n') {
+                flags = 0;
```

Comment:
0 is standard for no flags and it is shorter then anything else so I don't think we have to be extra semantic prudent here.

However I wonder if this is really needed. One `strlen` is not a performance issue, of course. However we produce tons of debug messages and we already spend lots of time in `strcmp` and `strlen`. So given this is for messages that you want to actually get rid of and are almost never useful, I think we can skip this patch.

Reply:
> However I wonder if this is really needed. One `strlen` is not a performance issue, of course. However we produce tons of debug messages and we already spend lots of time in `strcmp` and `strlen`. So given this is for messages that you want to actually get rid of and are almost never useful, I think we can skip this patch.

This is inside "if (level == LDB_DEBUG_TRACE)" that should not generally be used, that's why I think this is not a performance issue.

But indeed I don't mind dropping this patch.

Reply:
Removed this patch.